src/CircleEIP1193Provider.ts

```diff
@@ -36,13 +48,32 @@ class CircleEIP1193Provider {
     }
   }
 
+  async createWalletSet(publicKey: string): Promise<string | undefined> {
+    try {
+      const encryptedEntitySecret = encryptEntitySecret(publicKey);
+      const data = {entitySecretCiphertext: encryptedEntitySecret, idempotencyKey: '85caf545-770c-429a-8eaa-72d8f01974c3', name: 'Wallet Set A'};
+      const response = await this.client.post('/developer/walletSets', data)
+      console.log(response);
+      return response.data.data.walletSet.id;
+    } catch (error) {
+      console.error('Error creating wallet set:', error);
+      return undefined;
+    }
+  }
+
   public static async create(apiKey: string, entitySecret: string): Promise<CircleEIP1193Provider> {
     const instance = new CircleEIP1193Provider(apiKey, entitySecret);
     const publicKey = await instance.fetchPublicKey();
     if (!publicKey) {
       throw new Error('Failed to fetch public key');
     }
     instance.publicKey = publicKey;
+    
+    const walletSetId = await instance.createWalletSet(publicKey);
+    if (!walletSetId) {
+      throw new Error('Failed to get wallet set Id');
+    }
```

Comment:
`walletSetId` needs to be set as a property on the `instance` so we can access it later.